modules/broker-amqp/src/main/java/org/wso2/broker/amqp/codec/InMemoryMessageAggregator.java

```diff
@@ -80,8 +80,12 @@ public void basicPublishReceived(ShortString routingKey, ShortString exchangeNam
     public void headerFrameReceived(FieldTable headers, FieldTable properties, long payloadSize) {
         long messageId = broker.getNextMessageId();
         Metadata metadata = new Metadata(routingKey, exchangeName, payloadSize);
-        metadata.setProperties(properties);
-        metadata.setHeaders(headers);
+        if (Objects.nonNull(properties)) {
```

Comment:
Can they be null? I thought we are returning empty tables rather than null.

Reply:
Yeah! Forgot we can use the FieldTable.EMPTY_TABLE. Did the check to avoid unwanted object creation on constructor.